### ISSUE TITLE
Chinese garbled characters problem

### ISSUE THREAD

**Issue by ClarkWain** [TabbyML/tabby#691]
I run CodeLlama7B model on Tabby，I found that when it auto-completes Chinese comments, there is gibberish appearing. However, when I tried CodeLlama7B on HuggingFace, it seems to support Chinese.

What could be the reason for this discrepancy?

![chinese](https://github.com/TabbyML/tabby/assets/2370865/283ac56a-e4bf-4c10-8560-9e6ac9d03bfc)
![chinese2](https://github.com/TabbyML/tabby/assets/2370865/9ab834a4-fcd3-4e78-98fa-2cf33a2501c4)

**Comment by wsxiaoys** [comment]
Hi, please follow the bug report template to providing following information


> **Information about your version**
> Please provide output of `tabby --version`

> **Information about your GPU**
> Please provide output of `nvidia-smi`

**Comment by ClarkWain** [comment]
> Hi, please follow the bug report template and provide following information
> 
> > **Information about your version**
> > Please provide output of `tabby --version`
> 
> > **Information about your GPU**
> > Please provide output of `nvidia-smi`

```shell
./tabby --version
tabby 0.4.0
```


```shell
nvidia-smi
Fri Nov  3 10:15:21 2023       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.54.03              Driver Version: 535.54.03    CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA TITAN Xp                Off | 00000000:02:00.0 Off |                  N/A |
| 19%   35C    P0              61W / 250W |      0MiB / 12288MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   1  NVIDIA TITAN Xp                Off | 00000000:03:00.0 Off |                  N/A |
| 19%   30C    P0              61W / 250W |      0MiB / 12288MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   2  NVIDIA TITAN Xp                Off | 00000000:82:00.0 Off |                  N/A |
| 23%   22C    P8               8W / 250W |   8668MiB / 12288MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   3  NVIDIA TITAN Xp                Off | 00000000:83:00.0 Off |                  N/A |
| 21%   36C    P0              57W / 250W |      0MiB / 12288MiB |      1%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    2   N/A  N/A   2022650      C   /opt/tabby/bin/tabby                       8666MiB |
+---------------------------------------------------------------------------------------+
```

```shell
docker run \
  --gpus device=2 -p 8080:8080 -v /data1/weitianpeng/docker_main:/data \
  tabby040 \
  serve --model TabbyML/CodeLlama-7B --device cuda'
```

**Comment by wsxiaoys** [comment]
Could you give latest released candidate `0.5.0-rc.0` a try to see if it fixed the issue?

**Comment by ClarkWain** [comment]
![1](https://github.com/TabbyML/tabby/assets/2370865/775ab528-a2c3-46a6-9dc4-4ce714d0ff1b)

```shell
$ sudo docker run -it tabbyml/tabby --version
tabby 0.5.0
```

I've tried using version 0.5.0 as suggested, but the Chinese character encoding issue still exists. Could you please look into this further?

Thank you,

**Comment by wsxiaoys** [comment]
Could you verify with below curl command? This is what I get with `0.5.0`:
```
curl -X 'POST' \
  'http://localhost:8080/v1/completions' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "language": "python",
  "segments": {
    "prefix": "// 该函数",
    "suffix": "def fib(n):\n    "
  }
}'

{"id":"cmpl-0e8deef4-b9c2-44ed-93c3-b8546616c5ae","choices":[{"index":0,"text":"的作用是返回一个列表，列表中的元素是从0到n的斐波那契数列。\n"}]}
```

**Comment by ClarkWain** [comment]
```shell
curl -X 'POST'   'http://172.27.158.3:8080/v1/completions'   -H 'accept:
application/json'   -H 'Content-Type: application/json'   -d '{
  "language": "python",
  "segments": {
    "prefix": "// 该函数",
    "suffix": "def fib(n):\n    "
  }
}'
{"id":"cmpl-05707e7f-dc9b-4cf7-84b6-fbbe012e0180","choices":[{"index":0,"text":"的作用是返回一个列表，列表中的元素是从0 到n的斐波那契数列。\n"}]}
```


It seems that the results obtained with Curl are correct, but the results in the Android Studio are incorrect. Also, I have updated the plugin to the latest version.  



![2](https://github.com/TabbyML/tabby/assets/2370865/68312dd4-63a6-4486-895b-b79b76018df0)
--
![3](https://github.com/TabbyML/tabby/assets/2370865/b41cb9ed-efe6-4862-8136-8350d00735ba)
--
![4](https://github.com/TabbyML/tabby/assets/2370865/6a5aba2b-5b46-441b-b53f-4b50ec6d46fd)

--

I tried it on VSCode and there was no Chinese character encoding issue. The result is normal.
![5](https://github.com/TabbyML/tabby/assets/2370865/ba585785-7e98-43b5-820b-c9458eeedeec)

**Comment by wsxiaoys** [comment]
cc @icycodes to take a look to see if it's something specific to intellij platform (utf-8 related?)

**Comment by icycodes** [comment]
It appears to be a bug related to communication with the tabby-agent.

@ClarkWain, thanks for reporting this issue.
Could you please follow [this guide](https://tabby.tabbyml.com/docs/extensions/troubleshooting#want-to-deep-dive-via-logs) to enable debug logging and share the generated logs?

Additionally, it would be helpful if you could share your OS type and system locale info. (`cat /etc/default/locale` if you are using Ubuntu)

**Comment by ClarkWain** [comment]
![8](https://github.com/TabbyML/tabby/assets/2370865/0ea30201-f491-4d66-8b73-6e94da1fbea4)
Here is the log file:
[tabby-agent.log](https://github.com/TabbyML/tabby/files/13258681/tabby-agent.log)

Here is what I input:
![6](https://github.com/TabbyML/tabby/assets/2370865/64afc4c3-aad6-42e1-9f38-a8308c75d15b)

Here is the IDE settings:
![7](https://github.com/TabbyML/tabby/assets/2370865/aa511d95-a8b9-4b0f-964d-127e0026135c)

Here is system info:
![9](https://github.com/TabbyML/tabby/assets/2370865/5d579735-f7f8-4cfb-8829-ec3a5d09cc85)



![10](https://github.com/TabbyML/tabby/assets/2370865/427476b3-119b-46ee-9ba6-2b26b266a7ab)

**Comment by bitsz** [comment]
![image](https://github.com/TabbyML/tabby/assets/20807514/b0d70ef4-8b88-48db-829d-ff2124f27e6d)
一样的错误

**Comment by mjpcasey** [comment]
我有相同的问题，我看返回是中文来的，但在IDE上却是乱码，我想是不是需要改一下请求头部

**Comment by ShawnChen191** [comment]
如果修改了操作系统的编码，可以暂时解决乱码问题，但感觉并未从根本上解决
![image](https://github.com/TabbyML/tabby/assets/54101605/70d32327-e08f-4754-8bd3-162b44a90d7f)

**Comment by bitsz** [comment]
@ShawnChen191 你是对的

**Comment by 999jacky** [comment]
Help -> Etit Custom VM Options 
add `-Dfile.encoding=UTF-8`
Resolve for me

**Comment by ClarkWain** [comment]
I tried it, but it doesn't seem to work for me.

**Comment by ClarkWain** [comment]
> Help -> Etit Custom VM Options add `-Dfile.encoding=UTF-8` Resolve for me

I tried it, but it doesn't seem to work for me.

**Comment by icycodes** [comment]
This bug has been resolved in Tabby IntelliJ plugin v1.1.0. Please update your plugin to the latest version and give it a try.  

I will now mark this issue as closed. However, if you find that the issue still persists or if it does not work as intended, please feel free to reopen it.

**Comment by ClarkWain** [comment]
> This bug has been resolved in Tabby IntelliJ plugin v1.1.0. Please update your plugin to the latest version and give it a try.
> 
> I will now mark this issue as closed. However, if you find that the issue still persists or if it does not work as intended, please feel free to reopen it.

Thandks, it works.
